shiny/_template_utils.py

```diff
@@ -0,0 +1,150 @@
+import shutil
+import sys
+from pathlib import Path
+
+import questionary
+from questionary import Choice
+
+from ._custom_component_template_questions import (
+    install_js_dependencies,
+    isValidName,
+    updateComponentNameInTemplate,
+)
+
+
+def template_query():
+    # TODO: If we add additional menu levels we will need to modify this function so
+    # that the recursion picks up at the right level when the user clicks back.
+    template = questionary.select(
+        "Which template would you like to use?:",
+        choices=choicesArray(
+            [
+                ("Basic App", "basic-app"),
+                ("Express app", "express"),
+                ("Dashboard", "dashboard"),
+                ("Multi-page app with modules", "multi-page"),
+                ("Custom JavaScript Component", "js-component"),
+                ("Cancel", "cancel"),
+            ]
+        ),
+    ).ask()
+
+    # Define the control flow for the top level menu
+    # This is simple now but will get more complicated as we add more templates
+    flow_dispatch = {
+        "cancel": lambda: sys.exit(),
+        "js-component": lambda: jsComponentQuestions(),
+        "default": lambda: appTemplateQuestions(template),
+    }
+
+    # Call the appropriate function based on the user's choice
+    flow_dispatch.get(template, flow_dispatch["default"])()
```

Comment:
I'm curious about people's thoughts on this. It feels nice and extensible but also may be trying to be too clever. @schloerke @gshotwell

Reply:
Seems sensible to me